### PR TITLE
add arg builder to {set,get,delete}BucketNotification APIs

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -4840,14 +4840,47 @@ public String getBucketLifeCycle(GetBucketLifeCycleArgs args)
    * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
    * @throws XmlParserException thrown to indicate XML parsing error.
    */
+  @Deprecated
```
Comment (anjalshireesh) [inline api/src/main/java/io/minio/MinioClient.java:4843]: Good to add deprecation comment also with link to new method

File: api/src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -4922,12 +5005,50 @@ public void setBucketNotification(
...
       throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
           InternalException, InvalidBucketNameException, InvalidKeyException,
           InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
-    NotificationConfiguration notificationConfiguration = new NotificationConfiguration();
-    setBucketNotification(bucketName, notificationConfiguration);
+    setBucketNotification(
+        SetBucketNotificationArgs.builder()
+            .bucket(bucketName)
+            .config(new NotificationConfiguration())
+            .build());
```
Comment (anjalshireesh) [inline api/src/main/java/io/minio/MinioClient.java:5017]: Might be better to call `deleteBucketNotification` here

File: api/src/main/java/io/minio/MinioClient.java (hunk 1 above)
Comment (balamurugana) [reply api/src/main/java/io/minio/MinioClient.java:4966]: Done

File: api/src/main/java/io/minio/MinioClient.java (hunk 2 above)
Comment (balamurugana) [reply api/src/main/java/io/minio/MinioClient.java]: Done

**Pull request merged**